File: HSEvents.Web/client/models/school.models.ts
import { Address } from './address.models';
import { ContactInfo } from './other.models';

export class School {
    id: number;
    name: string;
    type: SchoolType;
    number: number;
    belongToUniversityDistrict: boolean;
    hasPriority: boolean;
    addresses: Address[];
    contacts: ContactPerson[];
}

export class SchoolType {
    id: number;
    name: string;
}

export class ContactPerson {
    id: number;
    appointment: string;
    contactInfo: ContactInfo;
}

File: Infrastructure/Repositories/Dto/SchoolModels.cs

using System.Collections.Generic;
using Domain;

namespace Infrastructure.Repositories.Dto
{
    public class SchoolDto
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public SchoolType Type { get; set; }
        public  int? Number { get; set; }
        public bool BelongToUniversityDistrict { get; set; }
        public bool HasPriority { get; set; }
        public List<AddressDto> Address { get; set; }
        public List<ContactPerson> Contacts { get; set; }
    }

}